assembl/graphql/schema.py

```diff
@@ -233,46 +431,54 @@ def resolve_video(self, args, context, info):
         )
 
     def resolve_img_url(self, args, context, info):
+        images = [
+            'https://framapic.org/CiOjTZIVLFgg/9Xr4dPUMDZJG.jpg',
+            'https://framapic.org/pF0R9jQM5Aof/MortpNh6czMP.jpg',
+            'https://framapic.org/3CL5o6dir9Vk/xQ0Uckbi7ckp.jpg',
+            'https://framapic.org/cKULJ8P6VLFr/Nojx87ovxtjC.jpg'
+        ]
         # TODO imgUrl
-        return ""
+        return random_sample(images, 1)[0]
 
 
 class Query(graphene.ObjectType):
     node = Node.Field()
-    posts = SQLAlchemyConnectionField(PostConnection)#, idea_id=graphene.ID())
-    # ideas = SQLAlchemyConnectionField(Idea)
-    thematics = graphene.List(Thematic, identifier=graphene.String())
+    posts = SQLAlchemyConnectionField(PostConnection, idea_id=graphene.ID())
+    ideas = SQLAlchemyConnectionField(Idea)
+    thematics = graphene.List(Thematic, identifier=graphene.String(required=True))
     # agent_profiles = SQLAlchemyConnectionField(AgentProfile)
 
     def resolve_ideas(self, args, context, info):
-        # TODO do we need to do a special query to only retrieve ideas
-        # associated to the RootIdea?
         connection_type = info.return_type.graphene_type  # this is IdeaConnection
         model = connection_type._meta.node._meta.model  # this is models.Idea
         query = get_query(model, context)
         discussion_id = context.matchdict['discussion_id']
-        query = query.filter(model.discussion_id == discussion_id
-            ).filter(model.hidden == False
-            ).filter(model.tombstone_condition())
-
+        discussion = models.Discussion.get(discussion_id)
+        root_idea_id = discussion.root_idea.id
+        descendants_query = model.get_descendants_query(
+            root_idea_id, inclusive=False)
+        query = query.filter(model.id.in_(descendants_query)
+            ).filter(model.hidden == False).order_by(model.id)
         # pagination is done after that, no need to do it ourself
         return query
 
     def resolve_posts(self, args, context, info):
         connection_type = info.return_type.graphene_type  # this is PostConnection
         model = connection_type._meta.node._meta.model  # this is models.PostUnion
-        query = get_query(model, context)
         discussion_id = context.matchdict['discussion_id']
-        query = query.filter(model.discussion_id == discussion_id
-            ).filter(model.hidden == False
-            ).filter(model.tombstone_condition()
-            ).filter(model.publication_state == models.PublicationStates.PUBLISHED)
-
-        # TODO filter posts for a specific idea
-        # idea_id = args.get('idea_id', None)
-        # if idea_id is not None:
-        #     id_ = int(Node.from_global_id(idea_id)[1])
-        #     query = query.filter(model.idea_content_links? idea_id? == id_)
+        idea_id = args.get('idea_id', None)
+        if idea_id is not None:
+            id_ = int(Node.from_global_id(idea_id)[1])
```

Comment:
Relay style Thematic:123 in base64. I don't use assembl style id at all in the graphql code.